src/main/java/de/tum/in/www1/artemis/repository/UserRepository.java

```diff
@@ -67,6 +67,16 @@
             + "(user.login like :#{#loginOrName}% or concat_ws(' ', user.firstName, user.lastName) like %:#{#loginOrName}%)")
     List<User> searchByLoginOrNameInGroup(@Param("groupName") String groupName, @Param("loginOrName") String loginOrName);
 
+    /**
+     * Gets users in a group by their registration number.
+     * @param groupName Name of group in which to search for users
+     * @param registrationNumbers Registration numbers of users
+     * @return found users that match the criteria
+     */
+    @EntityGraph(type = LOAD, attributePaths = { "groups" })
+    @Query("select user from User user where :#{#groupName} member of user.groups and " + "user.registrationNumber in :#{#registrationNumbers}")
```

Comment:
```suggestion
    @Query("select user from User user where :#{#groupName} member of user.groups and user.registrationNumber in :#{#registrationNumbers}")
```

this could be combined